File: 2.server/src/main/java/ru/vitaliy/petrov/server/forms/requests/documents/AdministrativeOffenceCaseProtocolUpdateRequest.java
package ru.vitaliy.petrov.server.forms.requests.documents;

import lombok.Data;

import javax.validation.constraints.NotEmpty;
import javax.validation.constraints.NotNull;
import java.sql.Date;
import java.sql.Time;

@Data
public class AdministrativeOffenceCaseProtocolUpdateRequest {

    @NotNull
    private final Long documentID;

    @NotNull
    private final Date updatedDateOfFill;

    @NotNull
    private final Time updatedTimeOfFill;

    @NotNull
    @NotEmpty
    private final String updatedPlaceOfFill;

    @NotNull
    @NotEmpty
    private final String updatedCulpritActualPlaceOfResidence;

    @NotNull
    @NotEmpty
    private final String updatedLawViolationInfo;

    @NotNull
    @NotEmpty
    private final String updatedCaseAdditionalInfo;

    @NotNull
    @NotEmpty
    private final String updatedPlaceAndTimeOfCaseExamination;

    @NotNull
    @NotEmpty
    private final String updatedChangedPlaceOfCaseExamination;

    @NotNull
    @NotEmpty
    private final String updatedExplanationsAndRemarksOfProtocol;

    @NotNull
    private final Long carAccidentEntityID;
}

File: 3.client/app/src/main/java/com/petrov/vitaliy/caraccidentapp/domain/models/requests/documents/AdministrativeOffenceCaseProtocolUpdateRequest.kt
package com.petrov.vitaliy.caraccidentapp.domain.models.requests.documents

import java.sql.Date
import java.sql.Time

data class AdministrativeOffenceCaseProtocolUpdateRequest(
    val documentID: Long,
    val updatedDateOfFill: Date,
    val updatedTimeOfFill: Time,
    val updatedPlaceOfFill: String,
    val updatedCulpritActualPlaceOfResidence: String,
    val updatedLawViolationInfo: String,
    val updatedCaseAdditionalInfo: String,
    val updatedPlaceAndTimeOfCaseExamination: String,
    val updatedChangedPlaceOfCaseExamination: String,
    val updatedExplanationsAndRemarksOfProtocol: String,
    val carAccidentEntityID: Long
)